gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/ContentValues.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+
+/** Contains details about one content id: key and the expired values. */
+public class ContentValues implements Serializable {
+
+  private static final long serialVersionUID = -1268447510930923804L;
+  private ContentKey contentKey;
+  private final Set<Content> expiredContents = new HashSet<>();
```

Comment:
I still think that `Content` should be replaced here with something like this:

```java 

  interface GcContentIdent {
    String getId();
  }
  @Value.Immutable
  interface IcebergTableGcIdent extends GcContentIdent {
    long getSnapshotId();
    static IcebergTableGcIdent fromContent(Content content) {
      ...
    }
  }
  @Value.Immutable
  interface IcebergViewGcIdent extends GcContentIdent {
    int getVersionId();
    static IcebergViewGcIdent fromContent(Content content) {
      ...
    }
  }
```

Currently, both `IcebergTable` and `IcebergView` are rather small objects and the "win" using the above pretty low.
But once we store the whole table/view-metadata in Nessie, those objects will become pretty big.

I'd be happy, if you can do this refactor in a follow-up.

Reply:
I want to keep it generic now, based on the need and advantages, I can do a follow up.